include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -358,58 +358,103 @@ def NonBlockingAssignOp : AssignOpBase<"nonblocking_assign"> {
 // Statements
 //===----------------------------------------------------------------------===//
 
-def None: I32EnumAttrCase<"None", 0, "none">;
-/// Transit from 0 to x/z/1, and from x/z to 1.
+// Any change on the input.
+def AnyChange: I32EnumAttrCase<"AnyChange", 0, "any">;
+// A transition from 0 to X/Z/1, or from X/Z to 1.
 def PosEdge: I32EnumAttrCase<"PosEdge", 1, "posedge">;
-/// Transit from 1 to x/z/0, and from x/z to 0.
+// A transition from 1 to X/Z/0, or from X/Z to 0.
 def NegEdge: I32EnumAttrCase<"NegEdge", 2, "negedge">;
-/// Include the negedge and posedge.
+// The combination of `PosEdge` and `NegEdge`.
 def BothEdges: I32EnumAttrCase<"BothEdges", 3, "edge">;
 
-def EdgeAtrr: I32EnumAttr<"Edge", "Edge kind",
-                          [None, PosEdge, NegEdge, BothEdges]>{
+def EdgeAttr: I32EnumAttr<"Edge", "Edge kind",
+                          [AnyChange, PosEdge, NegEdge, BothEdges]> {
   let cppNamespace = "circt::moore";
 }
 
-def EventOp : MooreOp<"wait_event", [
-  HasParent<"ProcedureOp">
+def WaitEventOp : MooreOp<"wait_event", [
+  RecursiveMemoryEffects,
```

Comment:
I don't think so, because you can write something like this:
```systemverilog
task foo(ref logic x);
  @(posedge x);
endtask

module Bar;
  logic y;
  always foo(y);
endmodule
```
which would probably lower to this:
```mlir
func.func @foo(%x: !moore.ref<l1>) {
  moore.wait_event {
    %0 = moore.read %x : <l1>
    moore.detect_event posedge %0 : l1
  }
}
hw.module @Bar() {
  %y = moore.variable : <l1>
  moore.procedure always {
    func.call @foo(%y) : (!moore.ref<l1>) -> ()
  }
}
```
I'm not entirely sure how we'd deal with this… probably some lowering pass on functions to make sure we can suspend in the middle of a function call 🤔

Reply:
Ugh forgot about tasks 😞 
That's a good question indeed.